src/mapgen_v7.cpp

```diff
@@ -513,8 +519,11 @@ int MapgenV7::generateBaseTerrain()
 		heightmap[index]       = surface_y;
 		ridge_heightmap[index] = surface_y;
 
-		if (surface_y > stone_surface_max_y)
-			stone_surface_max_y = surface_y;
+		if (surface_y > *stone_surface_max_y)
+			*stone_surface_max_y = surface_y;
+
+		if (surface_y < node_max.Y)
+			*generate_mountain = true;
```

Comment:
Instead of having this function /decide/ if a mountain should be generated, why doesn't it just report the lowest point generated in this chunk?  The logic of whether or not to generate the mountain could be outside of generateTerrain().